pkg/registry/pod/etcd/etcd.go

```diff
@@ -240,6 +258,21 @@ func (r *LogREST) Get(ctx api.Context, name string, opts runtime.Object) (runtim
 	if err != nil {
 		return nil, err
 	}
+
+	if location.Host == "" {
+		return nil, fmt.Errorf("Empty location.Host in %#v", location)
+	}
+
+	hostName := strings.Split(location.Host, ":")[0]
```

Comment:
Yes - we just get a prefix, and glue the port back in later on if we find DaemonEndpoints. This is one of those ugly things that were necessary to make it backward compatible.